extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -848,7 +866,17 @@ export class VariantAnalysisManager
       this.app.logger,
       "Cancelling variant analysis. This may take a while.",
     );
-    await cancelVariantAnalysis(this.app.credentials, variantAnalysis);
+    try {
+      await cancelVariantAnalysis(this.app.credentials, variantAnalysis);
+    } catch (e) {
+      if (variantAnalysis.status === VariantAnalysisStatus.Canceling) {
+        await this.onVariantAnalysisUpdated({
+          ...variantAnalysis,
+          status: VariantAnalysisStatus.InProgress,
+        });
```

Comment:
Why are we setting the status as `InProgress` here? I expected `Canceling`, but I might be getting confused with the logic 🙈

Reply:
We're in a code branch where we failed to cancel the variant analysis (for whatever reason), so it seemed that the right thing to do is to put it back to in progress so the user can try and cancel again.

Reply:
Ah I see, thanks!